File: llm_analysis_service/src/web/schemas/analysis.py
from pydantic import BaseModel

from src.services.models import (
    SummaryResult,
    KeyPointsResult,
    ActionItemsResult,
    FaqResult,
)


class SummaryResponse(BaseModel):
    """Response schema для POST /analysis/{recording_id}/summary."""

    status: str
    data: SummaryResult


class KeyPointsResponse(BaseModel):
    """Response schema для POST /analysis/{recording_id}/key-points."""

    status: str
    data: KeyPointsResult


class ActionItemsResponse(BaseModel):
    """Response schema для POST /analysis/{recording_id}/action-items."""

    status: str
    data: ActionItemsResult


class FaqResponse(BaseModel):
    """Response schema для POST /analysis/{recording_id}/faq."""

    status: str
    data: FaqResult

File: frontend/src/api/analysis.ts
import axios from 'axios'
import { useAuthStore } from '../stores/authStore'

export const analysisApi = axios.create({
  baseURL: '/api/analysis',
})

analysisApi.interceptors.request.use((config) => {
  const token = useAuthStore.getState().token
  if (token) {
    config.headers.Authorization = `Bearer ${token}`
  }
  return config
})

analysisApi.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response?.status === 401) {
      useAuthStore.getState().logout()
      window.location.href = '/login'
    }
    return Promise.reject(error)
  }
)

// Types
export interface SummaryData {
  summary: string
  topics: string[]
}

export interface KeyPoint {
  point: string
  speaker: string
}

export interface ActionItem {
  action: string
  assignee: string
  context: string
}

export interface FaqItem {
  question: string
  answer: string
}

export interface ChatAnswer {
  answer: string
  quote: string
}

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
  timestamp: string
}

export interface TranscriptSegment {
  speaker: string
  start: number
  end: number
  text: string
}

export type AnalysisType = 'summary' | 'key-points' | 'action-items' | 'faq'

// API functions
export function runAnalysis(recordingId: string, type: AnalysisType) {
  return analysisApi.post(`/analysis/${recordingId}/${type}`)
}

export function sendChatMessage(recordingId: string, question: string) {
  return analysisApi.post(`/analysis/${recordingId}/chat`, { question })
}

export function getChatHistory(recordingId: string) {
  return analysisApi.get(`/analysis/${recordingId}/chat/history`)
}

export function getTranscript(recordingId: string) {
  return analysisApi.get(`/analysis/recordings/${recordingId}/transcript`)
}
